doc/notes/cost-model-representations/cost-model-representations.md

```diff
@@ -0,0 +1,216 @@
+# Representations of the Plutus Core cost model
+
+The `plutus` codebase uses several different representations of the Plutus Core
+cost model.  It can be somewhat difficult to keep track of what all of these are
+used for; this document attempts to give an overview to clarify matters.  An
+essential point is that if you're using `mkEvaluationContext` then cost model
+parameters much be provided in the correct order, otherwise script costs
+will be inaccurate: see [`EvaluationContext`](#evaluationcontext) below.
```

Comment:
e.g. this information is crucial for users of `mkEvaluationContext`, so it should probably be discoverable there.

Reply:
It does already [say this](https://github.com/input-output-hk/plutus/blob/e098652b6985c6ae52477e2f0931fe2364732aaa/plutus-ledger-api/src/PlutusLedgerApi/V1/EvaluationContext.hs#L22):
> The input is a list of integer values passed from the ledger and
are expected to appear in correct order.

I'll expand on this and some of the other comments, but I think that a lot of the content of this document is already in the Haddock, but spread around in such a way that it's difficult to see the big picture.